### PR TITLE
Improve CodeGenerator logging

### PR DESCRIPTION
**Pull request by zurk**
Helps to catch strange behavior

### CODE REVIEW

File: lookout/style/format/code_generator.py [hunk 1]
```diff
@@ -46,11 +49,16 @@ def __init__(self,
...
         :param verbose: Run code with additional verbose debug logging.
+        :param url: Repository url if applicable. Useful for more informative warning messages.
+        :param commit: Commit hash if applicable. Useful for more informative warning messages.
+
         """
         self.feature_extractor = feature_extractor
         self.change_locally = change_locally
         self.skip_errors = skip_errors
         self.verbose = verbose
+        self.url = url if url else "<NA>"
```
Comment (m09) [inline lookout/style/format/code_generator.py]: `"<NA>"` should be the default argument, not `None`.

File: lookout/style/format/code_generator.py [hunk 2]
```diff
@@ -46,11 +49,16 @@ def __init__(self,
...
+        self.commit = commit if commit else "<NA>"
```
Comment (m09) [inline lookout/style/format/code_generator.py]: same

File: lookout/style/format/code_generator.py [hunk 3]
```diff
@@ -185,6 +195,7 @@ def _iterate_vnodes(self, vnodes, vnodes_y, y_pred):
                 y_index += 1
 
     def _error_handler(self, msg):
+        msg += " url: %s, commit: %s." % (self.url, self.commit)
         msg += (" Skipping." if self.skip_errors else
                 "Set skip_errors=True to skip faulty suggestions.")
```
Comment (m09) [inline lookout/style/format/code_generator.py]: lacks a leading space

File: lookout/style/format/code_generator.py [hunk 4]
```diff
@@ -185,6 +195,7 @@ def _iterate_vnodes(self, vnodes, vnodes_y, y_pred):
                 y_index += 1
 
     def _error_handler(self, msg):
+        msg += " url: %s, commit: %s." % (self.url, self.commit)
```
Comment (m09) [inline lookout/style/format/code_generator.py]: both additions are mergeable into one (with format string). It will be more readable.

**Comment by zurk** [comment]
Thanks! I also note that I named a function as `_error_handler` and it is better to start from verb, so change it to `_handle_error`

**Pull request merged**